### ISSUE TITLE
details on updating parameters

### ISSUE THREAD

**Issue by jizongFox** [AlanChou/Deep-Co-Training-for-Semi-Supervised-Image-Recognition#1]
Hi Zhou, 
Very happy to see your repo. I am noticing something and want to ask for the details.

The main idea of adversarial loss for two view is to let a hard example of model1 to be easy for model2. So I reduce the loss function KL(predict_model2(adversarial image targeted at model1), predict_model1(image) ). If the two models are the same, `predict_model1(image)` should perform well than `predict_model2(adversarial image targeted at model1)`. I would wonder why there is no .detach() operation for the target tensor? Without detach(), you would reduce the performance of easy example.

**Comment by AlanChou** [comment]
Hi,

I suppose your question about "detach() operation for the target tensor" is referred to the generation of the pseudo labels of the unlabeled data. I checked my code again and I realized that I made a mistake. A detach operation is indeed needed as you mention. Thank you so much for your correction. I will fix it as soon as possible and come up with a new results.

As for the first part of your question, are you referring to that you reduced "loss_diff" from a cross entropy form to KL? I was just simply following the author's descriptions in the paper. I'm curious of how much gain you're able to get by changing that? 

If I get your wrong, please let me know.

I'm very happy to see suggestions. Thank you again for your comments. More discussions are welcomed.

**Comment by jizongFox** [comment]
For me even with detach it doesn’t work. Could you please inform me if that works for you?

**Comment by AlanChou** [comment]
What do you mean by "not working"? Do you mean that the accuracy isn't as reported in the paper? May I ask what's the best results you got so far? If it works for me, I will surely inform you.

**Comment by jizongFox** [comment]
Hi Chou, I am working on different problems. I run some proof-of-concept experiments based on your work to see if it works in my case. 
Thanks, 
Jizong
